spec/integration/outputs/retry_spec.rb

```diff
@@ -11,146 +11,143 @@
   let(:max_retries) { 3 }
 
   def mock_actions_with_response(*resp)
-    LogStash::Outputs::Elasticsearch::Protocols::HTTPClient
+    LogStash::Outputs::Elasticsearch::Protocols::TransportClient
```

Comment:
good catch, fixed!

On Thu, Aug 20, 2015 at 2:36 PM, Tal Levy <notifications@github.com> wrote:

> In spec/integration/outputs/retry_spec.rb
> <https://github.com/logstash-plugins/logstash-output-elasticsearch_native/pull/1#discussion_r37571763>
> :
>
> > @@ -11,146 +11,143 @@
> >    let(:max_retries) { 3 }
> >
> >    def mock_actions_with_response(*resp)
> > -    LogStash::Outputs::Elasticsearch::Protocols::HTTPClient
> > +    LogStash::Outputs::Elasticsearch::Protocols::TransportClient
>
> 2 things.
>
>    1.
>
>    I don't fully understand why, but mocking both Node and Transport
>    Clients caused problems to me in the past. Since Transport client is of
>    type NodeClient, mocking the NodeClient sufficed.
>    2.
>
>    we might as well update to the newer convention in rspec for
>    any_instance mocking like so:
>
> allow_any_instance_of(LogStash::Outputs::Elasticsearch::Protocols::NodeClient).to receive(:bulk).and_return(*resp)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/logstash-plugins/logstash-output-elasticsearch_native/pull/1/files#r37571763>
> .
>